runtime/src/bank.rs

```diff
@@ -3374,7 +3374,16 @@ impl Bank {
                     let mut account = self
                         .get_account_with_fixed_root(&pubkey)
                         .unwrap_or_default();
-                    account.lamports += rent_to_be_paid;
+                    if account.checked_add_lamports(rent_to_be_paid).is_err() {
```

Comment:
On err do we want to incinerate the entire rent amount or just the amount that overflowed?

Reply:
Here is what @mvines said on slack to me:

> checked_add() seems great.  But if it returns None (nothing happened), then the payment gets added to the balance that's transferred to the incinerator address

Reply:
If lamports overflows here there is obviously a huge problem in general so maybe it doesn't matter if we add what we can to the account.  But it does seem weird to incinerate the entire `rent_to_be_paid` if some of it could be paid to the account.  Don't feel strongly either way.